src/library.cpp

```diff
@@ -143,27 +143,35 @@ Book& Library::getBookByPath(const std::string& path)
 
 std::shared_ptr<Reader> Library::getReaderById(const std::string& id)
 {
-  try {
-    return m_readers.at(id);
-  } catch (std::out_of_range& e) {}
+  if (m_readers.count(id)) {
+    return m_readers[id];
+  } else if (m_archives.count(id)) {
+    auto reader = make_shared<Reader>(m_archives[id]);
+    m_readers[id] = reader;
+    return reader;
+  }
```

Comment:
Is there an advantage of this technique? or is it better coding style?

Reply:
###### `at(key)` vs `[key]` :
`[key]` is stricky as if the key doesn't exist in the map, it will be created (with a nullptr shared_ptr here).
`at(key)` will throw a exception if the key is not found.
Because `[key]` create the entry, we must search if the key exist before using it (what you do).

###### `at(key)` vs `count(key)`+`[key]` :
With `at(key)` you do only one lookup. Either you find it and you get the entry, or not.
With the `count(key)`+`[key]` you need to do two lookup. On for the count, and then another one the get the entry.

On top of that, there is some kind of different semantics :
With a count, you means "If there is a key in the map, do something (which is return the entry)."
With the try/at you means "I want to return the entry, handling the case there is no entry in the map."

The second is a bit more direct and carry better what is your intention.